inlang/source-code/sdk/src/loadProject.ts

```diff
@@ -329,24 +327,10 @@ const loadSettings = async (args: {
 	settingsFilePath: string
 	nodeishFs: NodeishFilesystemSubset
 }) => {
-	const { data: settingsFile, error: settingsFileError } = await tryCatch(
-		async () => await args.nodeishFs.readFile(args.settingsFilePath, { encoding: "utf-8" })
-	)
-	if (settingsFileError)
-		throw new ProjectSettingsFileNotFoundError({
-			cause: settingsFileError,
-			path: args.settingsFilePath,
-		})
-
-	const json = tryCatch(() => JSON.parse(settingsFile!))
-
-	if (json.error) {
-		throw new ProjectSettingsFileJSONSyntaxError({
-			cause: json.error,
-			path: args.settingsFilePath,
-		})
-	}
-	return parseSettings(json.data)
+	// caller should handle errors
+	const settingsFile = await args.nodeishFs.readFile(args.settingsFilePath, { encoding: "utf-8" })
+	const json = JSON.parse(settingsFile)
+	return parseSettings(json)
```

Comment:
The UX of errors seems to be negatively impacted by this change. 

1. an app won't inform the user anymore that a settings file has not been found
2. an app won't inform the user anymore that the settings file contains a syntax error

Both errors were introduced after inlang users ran into them and did not understand the underlying error. Hence, we wrapped them to provide a better error message. 

Please elaborate on why this change is beneficial/required.

Reply:
ok - I reverted the JSON parse error so that it includes the settings file path in the error message. You're right, that's helpful for UX.

I think we should allow the readfile error to pass through to users, because the wrapper doesn't add additional information, and based on @janfjohannes experience last week, the wrapper can actually prevent lix failures from being reported accurately. 

before:
```
$ p inlang validate --project ./project.inlang/
ℹ 🔎 Validating the inlang project...                                                                                                  12:05:53 PM

 ERROR  The file at "/Users/jldec/inlang/load-test/project.inlang/settings.json" could not be read. Does the file exists?               12:05:53 PM

  at loadSettings (node_modules/.pnpm/@inlang+cli@2.12.6/node_modules/@inlang/cli/dist/main.js:98230:11)
```

with this PR
```
$ p inlang validate --project project.inlang/
ℹ 🔎 Validating the inlang project...                                                                                                  12:44:00 PM

[12:44:01 PM]  ERROR  ENOENT: no such file or directory, open '/Users/jldec/inlang/monorepo/inlang/source-code/paraglide/paraglide-js/example/project.inlang/settings.json'
```

Reply:
@jldec feel free to call me

Reply:
The CLI is not logging `error.cause`. That's a bug in the CLI. There seems little benefit in removing concrete errors in favor of generic one's just because one app is not logging correctly. 

`error.cause` has been introduced specifically to re-throw with more concrete error messages https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause